codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -572,42 +573,70 @@ private class ServerHttpProtocolImplGenerator(
-                pathBindings.forEach {
-                    val deserializer = generateParsePercentEncodedStrFn(it)
+            for (index in httpTrait.uri.segments.indices) {
+                val segment = httpTrait.uri.segments[index]
+                val name = segment.content
+                val binding = pathBindings.find { it.memberName == name }
+                val inputStringPrefix = if (index == httpTrait.uri.segments.size - 1) { "_" } else { "" }
+                val bindingPrefix = if (!(binding != null && segment.isLabel)) { "_" } else { "" }
+                rustTemplate(
+                    """
+                    let (input_string, _tag) = #{Nom}::bytes::complete::tag::<_, &str, #{Nom}::error::Error<&str>>("/")(input_string).unwrap();
+                    """.trimIndent(),
+                    *codegenScope
+                )
+                if (segment.isGreedyLabel) {
+                    val end = httpTrait.uri.segments.slice((index + 1) until httpTrait.uri.segments.size).joinToString(prefix = "/", separator = "/")
+                    if (end.length > 1)
+                        rustTemplate(
+                            """
+                            let (${bindingPrefix}m, ${inputStringPrefix}input_string) = match input_string.rfind(${end.dq()}) {
+                                Some(n) => (&input_string[..n], &input_string[n..]),
+                                None => (input_string, "")
+                            };
+                            """.trimIndent()
+                        )
+                    else
+                        rustTemplate(
+                            """
+                            let ${bindingPrefix}m = input_string;
+                            let ${inputStringPrefix}input_string = "";
+                            """.trimIndent()
+                        )
+                } else if (segment.isLabel) {
                     rustTemplate(
                         """
-                        if let Some(m) = captures.name("${it.locationName}") {
-                            input = input.${it.member.setterName()}(
-                                #{deserializer}(m.as_str())?
-                            );
-                        }
+                        let (${inputStringPrefix}input_string, ${bindingPrefix}m) = #{Nom}::branch::alt::<_, &str, #{Nom}::error::Error<&str>, _>((
+                            #{Nom}::bytes::complete::take_until1("/"),
+                            #{Nom}::combinator::rest))(input_string).unwrap();
+                        """.trimIndent(),
+                        *codegenScope
+                    )
+                } else {
+                    rustTemplate(
+                        """
+                        let (input_string, ${bindingPrefix}m) = #{Nom}::bytes::complete::tag::<_, &str, #{Nom}::error::Error<&str>>(${segment.content.dq()})(input_string).unwrap();
```

Comment:
```suggestion
                        let (${inputStringPrefix}input_string, ${bindingPrefix}m) = #{Nom}::bytes::complete::tag::<_, &str, #{Nom}::error::Error<&str>>(${segment.content.dq()})(input_string).unwrap();
```

I think clippy is failing in CI because of this.